cmd/dlt/oidcconfig/cmd.go

```diff
@@ -0,0 +1,230 @@
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package oidcconfig
+
+import (
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws/arn"
+	"github.com/briandowns/spinner"
+	"github.com/spf13/cobra"
+	"github.com/zgalor/weberr"
+
+	"github.com/openshift/rosa/pkg/arguments"
+	"github.com/openshift/rosa/pkg/aws"
+	awscb "github.com/openshift/rosa/pkg/aws/commandbuilder"
+	"github.com/openshift/rosa/pkg/interactive"
+	"github.com/openshift/rosa/pkg/interactive/confirm"
+	"github.com/openshift/rosa/pkg/rosa"
+)
+
+var Cmd = &cobra.Command{
+	Use:     "oidc-config",
+	Aliases: []string{"oidconfig"},
+	Short:   "Delete OIDC Config",
+	Long:    "Cleans up OIDC config base on secret ARN.",
+	Example: `  # Delete OIDC config secret ARN that has been supplied
+	rosa delete oidc-config --oidc-private-key-secret-arn <oidc_private_key_secret_arn>`,
+	Hidden: true,
+	Run:    run,
+}
+
+const (
+	//nolint
+	OidcPrivateKeySecretArnFlag = "oidc-private-key-secret-arn"
+
+	prefixForPrivateKeySecret = "rosa-private-key-"
+)
+
+var args struct {
+	oidcPrivateKeySecretArn string
+	region                  string
+}
+
+func init() {
+	flags := Cmd.Flags()
+
+	flags.StringVar(
+		&args.oidcPrivateKeySecretArn,
+		OidcPrivateKeySecretArnFlag,
+		"",
+		"AWS Secrets Manager ARN for identification of config",
+	)
+
+	aws.AddModeFlag(Cmd)
+
+	interactive.AddFlag(flags)
+	confirm.AddFlag(flags)
+}
+
+func run(cmd *cobra.Command, argv []string) {
+	r := rosa.NewRuntime().WithAWS()
+	defer r.Cleanup()
+
+	mode, err := aws.GetMode()
+	if err != nil {
+		r.Reporter.Errorf("%s", err)
+		os.Exit(1)
+	}
+
+	// Get AWS region
+	region, err := aws.GetRegion(arguments.GetRegion())
+	if err != nil {
+		r.Reporter.Errorf("Error getting region: %v", err)
+		os.Exit(1)
+	}
+	args.region = region
+
+	// Determine if interactive mode is needed
+	if !interactive.Enabled() && !cmd.Flags().Changed("mode") {
+		interactive.Enable()
+	}
+
+	if interactive.Enabled() {
+		mode, err = interactive.GetOption(interactive.Input{
+			Question: "OIDC config creation mode",
+			Help:     cmd.Flags().Lookup("mode").Usage,
+			Default:  aws.ModeAuto,
+			Options:  aws.Modes,
+			Required: true,
+		})
+		if err != nil {
+			r.Reporter.Errorf("Expected a valid OIDC provider creation mode: %s", err)
+			os.Exit(1)
+		}
+	}
+
+	oidcConfigInput := buildOidcConfigInput(r)
+	oidcConfigStrategy, err := getOidcConfigStrategy(mode, oidcConfigInput)
+	if err != nil {
+		r.Reporter.Errorf("%s", err)
```

Comment:
Same for this, `%w`?

Reply:
Same as https://github.com/openshift/rosa/pull/1057#discussion_r1099081924